packages/react/ds-app-launchpad/src/ui/FileTree/common/Node/Provider.tsx

```diff
@@ -0,0 +1,232 @@
+
+  const [isExpanded, setIsExpanded] = useState(
+    nodeOptions.nodeType === "folder" && nodeOptions.defaultOpen,
+  );
+  const path = useMemo(() => `${parentPath}/${name}`, [parentPath, name]);
+  const isSelected = useMemo(
+    () => selectedFile?.path === path,
+    [selectedFile, path],
+  );
+
+  const matchesQuery = useCallback((query: string, target: string) => {
+    return target.toLowerCase().includes(query.toLowerCase());
+  }, []);
+
+  const fileMatchesSearchQuery = useCallback(
+    (node: FileTreeData) => {
+      if (!searchQuery) return true;
+      // check if path matches in case of files
+      if (node.type === "file") {
+        return matchesQuery(searchQuery, node.name);
+      }
+      // check if any child/grandchild matches in case of folders
+      return (
+        matchesQuery(searchQuery, node.name) ||
+        node.children?.some(fileMatchesSearchQuery)
+      );
+    },
+    [searchQuery, matchesQuery],
+  );
+
+  const shouldHideNode = useMemo(() => {
+    if (!searchQuery) return false;
+    return !fileMatchesSearchQuery(nodeRef.current);
+  }, [searchQuery, fileMatchesSearchQuery]);
+  const handleClick: React.MouseEventHandler<HTMLLIElement> = useCallback(
+    (_event) => {
+      switch (nodeOptions.nodeType) {
+        case "folder":
+          if (expandable) {
+            setIsExpanded((prev) => !prev);
+          }
+          break;
+        case "file":
+          onSelectFile?.({
+            name,
+            path,
+            type: nodeOptions.nodeType,
+          });
+          break;
+      }
+    },
+    [nodeOptions.nodeType, onSelectFile, name, path, expandable],
+  );
+
+  const handleKeyDown: React.KeyboardEventHandler<HTMLLIElement> = (event) => {
+    if (event.ctrlKey || event.altKey || event.shiftKey || event.metaKey)
+      return;
+    if (event.key.match(/^[a-zA-Z0-9]$/)) {
+      focusNextSiblingCharacter(event.key);
+      event.preventDefault();
+    }
+    switch (event.key) {
+      case "ArrowRight":
+        if (!isExpanded && nodeOptions.nodeType === "folder" && expandable) {
+          setIsExpanded(true);
+          event.preventDefault();
+        } else {
+          focusNextNode("down");
+          event.preventDefault();
+        }
+        break;
+      case "ArrowLeft":
+        if (isExpanded && nodeOptions.nodeType === "folder" && expandable) {
+          setIsExpanded(false);
+          event.preventDefault();
+        } else {
+          focusNextNode("up");
+          event.preventDefault();
+        }
+        break;
+      case "ArrowDown":
+        focusNextNode("down");
+        event.preventDefault();
+
+        break;
+      case "ArrowUp":
+        focusNextNode("up");
+        event.preventDefault();
+        break;
+      case "Home":
+        focusEndNode("start");
+        event.preventDefault();
+        break;
+      case "End":
+        focusEndNode("end");
+        event.preventDefault();
+        break;
+      case "Enter":
```

Comment:
I think something may be off with the "ENTER" key handling here. When I use the keyboard to navigate your file tree, and press "ENTER" on a node, it is not selected. I'd expect that a file would be selected and a folder would be expanded/collapsed when I press enter.

Reply:
The key `Enter` is responsible for selecting a node.
In my case I only need selectable files, I have no use cases for selecting folders so I limited selection to files only.
Which results in Hitting Enter on folder has no effect. For expansion/collapse the keys `ArrowLeft` and `ArrowRight` are  used.